diff --git a/node_modules/@netlify/plugin-nextjs/index.js b/node_modules/@netlify/plugin-nextjs/index.js
index 9b2b707..e15b622 100644
--- a/node_modules/@netlify/plugin-nextjs/index.js
+++ b/node_modules/@netlify/plugin-nextjs/index.js
@@ -3,31 +3,28 @@ const path = require('path')
 const util = require('util')
 const makeDir = require('make-dir')
 const findUp = require('find-up')
-
+const getNetlifyFunctionName = require("next-on-netlify/lib/helpers/getNetlifyFunctionName")
+const AdmZip = require('adm-zip');
 const validateNextUsage = require('./helpers/validateNextUsage')
 const doesNotNeedPlugin = require('./helpers/doesNotNeedPlugin')
-
 const pWriteFile = util.promisify(fs.writeFile)
 
+const DEFAULT_FUNCTIONS_SRC = 'netlify/functions'
+
 // * Helpful Plugin Context *
 // - Between the prebuild and build steps, the project's build command is run
 // - Between the build and postbuild steps, any functions are bundled
-
 module.exports = {
   async onPreBuild({ netlifyConfig, packageJson, utils }) {
     const { failBuild } = utils.build
-
     validateNextUsage(failBuild)
-
     const hasNoPackageJson = Object.keys(packageJson).length === 0
     if (hasNoPackageJson) {
       return failBuild('Could not find a package.json for this project')
     }
-
     if (await doesNotNeedPlugin({ netlifyConfig, packageJson, utils })) {
       return
     }
-
     const nextConfigPath = await findUp('next.config.js')
     if (nextConfigPath === undefined) {
       // Create the next config file with target set to serverless by default
@@ -48,17 +45,36 @@ module.exports = {
     if (await doesNotNeedPlugin({ netlifyConfig, packageJson, utils })) {
       return
     }
-
     console.log(`** Running Next on Netlify package **`)
-
     await makeDir(PUBLISH_DIR)
-
     // We cannot load `next-on-netlify` (which depends on `next`) at the
     // top-level because we validate whether the site is using `next`
     // inside `onPreBuild`.
     const nextOnNetlify = require('next-on-netlify')
     nextOnNetlify({ functionsDir: FUNCTIONS_SRC, publishDir: PUBLISH_DIR })
   },
-}
+  async onPostBuild({ constants }) {
+    // loads the full Next config, including defaults
+    const { PHASE_PRODUCTION_BUILD } = require('next/constants')
+    const { default: loadConfig } = require('next/dist/next-server/server/config')
+    const config = loadConfig(PHASE_PRODUCTION_BUILD, path.resolve('.'));
     
-const DEFAULT_FUNCTIONS_SRC = 'netlify/functions'
+    // any functions defined in the config need special handling
+    for (const name in (config.functions || {})) {
+      const includes = config.functions[name].includeDir;
+
+      console.log("Processing includeFiles for ", name, includes)
+
+      const zipName = path.join(constants.FUNCTIONS_DIST, getNetlifyFunctionName(name) + ".zip")
+      const zip = new AdmZip(zipName)
+      if (fs.lstatSync(includes).isDirectory()) {
+        // we add the files at the root of the ZIP because process.cwd()
+        // points to `/` in serverless functions
+        zip.addLocalFolder(includes, includes);
+        console.log(`Added ${includes} to ${zipName}`);
+      }
+
+      zip.writeZip(zipName)
+    }
+  }
+}
